actors/evm/src/interpreter/instructions/lifecycle.rs

```diff
@@ -1,17 +1,161 @@
-use crate::interpreter::address::EthAddress;
+use fil_actors_runtime::ActorError;
+use fil_actors_runtime::EAM_ACTOR_ADDR;
+use fvm_ipld_encoding::{strict_bytes, tuple::*, RawBytes};
+use fvm_shared::MethodNum;
+use fvm_shared::{address::Address, econ::TokenAmount};
+use serde_tuple::{Deserialize_tuple, Serialize_tuple};
+
+use crate::interpreter::stack::Stack;
+use crate::interpreter::{address::EthAddress, U256};
+use crate::state::State;
+
+use super::memory::{get_memory_region, MemoryRegion};
 use {
     crate::interpreter::{ExecutionState, StatusCode, System},
     fil_actors_runtime::runtime::Runtime,
     fvm_ipld_blockstore::Blockstore,
 };
 
+pub const CREATE_METHOD_NUM: u64 = 2;
+pub const CREATE2_METHOD_NUM: u64 = 3;
+
+#[derive(Serialize_tuple, Deserialize_tuple, Clone)]
+pub struct CreateParams {
+    #[serde(with = "strict_bytes")]
+    pub code: Vec<u8>,
+    pub nonce: u64,
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple, Clone)]
+pub struct Create2Params {
+    #[serde(with = "strict_bytes")]
+    pub code: Vec<u8>,
+    pub salt: [u8; 32],
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple, Debug, Clone, Copy, PartialEq, Eq)]
+pub struct EamReturn {
```

Comment:
I'd just call this `CreateReturn` and alias it (to be consistent with everything else).